### PR TITLE
fix: trivydb update now uses task scheduler

### PR DESCRIPTION
**Pull request by aokirisaki**
**What type of PR is this?**

**Which issue does this PR fix**:
#1127

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/extensions/extension_search.go [hunk 1]
```diff
@@ -51,30 +59,94 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
...
+func newTrivyTask(interval time.Duration, cveInfo cveinfo.CveInfo,
+	generator *trivyTaskGenerator, log log.Logger,
+) *trivyTask {
+	return &trivyTask{interval, cveInfo, generator, time.Second, log}
+}
+
+func (trivyT *trivyTask) DoWork() error {
+	trivyT.generator.SetTrivyGeneratorStatus(statusRunning)
+	trivyT.log.Info().Msg("updating the CVE database")
+	time.Sleep(trivyT.waitTime)
```
Comment (andaaron) [inline pkg/extensions/extension_search.go:126]: This is actually not OK. we need to wait before generating the task. If we wait here, we block one of the goroutines of the scheduler. Not sure where's the correct place to put this. @Andreea-Lupu any thoughts?

File: pkg/extensions/extension_search.go [hunk 2]
```diff
@@ -51,30 +62,105 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
...
+	gen.waitTime = 0
+	gen.lock.Unlock()
+}
 
-		time.Sleep(updateInterval)
+type trivyTask struct {
+	interval  time.Duration
+	cveInfo   cveinfo.CveInfo
+	generator *trivyTaskGenerator
+	waitTime  time.Duration
```
Comment (andaaron) [inline pkg/extensions/extension_search.go:133]: This should no longer be needed here.

File: pkg/extensions/extension_search.go [hunk 3]
```diff
@@ -51,30 +62,105 @@ func EnableSearchExtension(config *config.Config, storeController storage.StoreC
...
+	interval     time.Duration
+	cveInfo      CveInfo
+	log          log.Logger
+	status       state
+	waitTime     time.Duration
+	lastTaskTime time.Time
+	lock         *sync.Mutex
+}
+
+func (gen *trivyTaskGenerator) SetTrivyGeneratorStatus(status state) {
```
Comment (andaaron) [inline pkg/extensions/extension_search.go:89]: Not really sure we need the setter/getter anymore, since we don't lock here anymore.

**Review by andaaron — APPROVED** [review]
Looks good!

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**